service/service.go

```diff
@@ -188,14 +193,17 @@ func (s *Service) DeleteUser(ctx context.Context, req *pb.UserRequest) (*pb.User
 	}
 
 	// validate user id
-	uuid := user.GetUuid()
-	if err := validateUUID(uuid); err != nil {
+	if err := validateUUID(user.GetUuid()); err != nil {
 		logger.Error("DeleteUser validating uuid: ", err.Error())
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
+	lock, _ := uuidMapLocker.LoadOrStore(user.GetUuid(), &sync.RWMutex{})
+	lock.(*sync.RWMutex).Lock()
+	defer lock.(*sync.RWMutex).Unlock()
+
 	// check uuid exists
-	exists, err := checkUserExists(uuid)
+	exists, err := checkUserExists(user.GetUuid())
 	if err != nil {
```

Comment:
if it does not exists, this means that we need to remove it from our cache `uuidMapLocker` to save resources or memory, because we allocated a Mutex for it. this can be a security issue.